src/Microsoft.AspNetCore.Authorization.Policy/AuthenticationPolicy.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    /// <summary>
+    /// AuthenticationPolicy assigns a set of schemes for all of the authentication actions.
+    /// </summary>
+    public class AuthenticationPolicy
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="authenticateSchemes"></param>
+        /// <param name="challengeSchemes"></param>
+        /// <param name="forbidSchemes"></param>
+        public AuthenticationPolicy(IEnumerable<string> authenticateSchemes,
+            IEnumerable<string> challengeSchemes,
+            IEnumerable<string> forbidSchemes)
+        {
+            AuthenticateSchemes = authenticateSchemes ?? new string[0];
+            ChallengeSchemes = challengeSchemes ?? new string[0];
+            ForbidSchemes = forbidSchemes ?? new string[0];
+        }
+
+        /// <summary>
+        /// The authentication schemes that should be used to construct the ClaimsPrincipal during authentication.
+        /// </summary>
+        public IEnumerable<string> AuthenticateSchemes { get; }
+
+        /// <summary>
+        /// The authentication schemes that should be challenged. 
+        /// </summary>
+        public IEnumerable<string> ChallengeSchemes { get; }
+
+        /// <summary>
+        /// The authentication schemes that should be forbidden.
+        /// </summary>
+        public IEnumerable<string> ForbidSchemes { get; }
+
+        // REVIEW: Do we really need SignIn/SignOut schemes?
```

Comment:
We need some way for remote providers to resolve their correct sign-in scheme. Or we make them specify it in their options, but it seems odd to leave out.

Reply:
Remote providers don't need to change from what they are doing today, they require a sign in scheme, it will use the default if not specified, we could have that use the default policy if specified as a small change, but I don't think adding a SignInPolicy really helps since its only using the SignInScheme

Reply:
Ok so if we are going to switch the Remote providers to use the new `IAuthentication,PolicyEvaluator` to SignIn, then we'll need to add a SignInPolicy setting which I guess would be used if SignInScheme isn't specified?  Gets confusing fast :)